### PR TITLE
Feat/ne 14

### PR DESCRIPTION
**Pull request by SergeyMushta**

### CODE REVIEW

File: contracts/ERC1155NftCustom.sol [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+    }
+
+    function transferByOwner(
+        address from,
+        address to,
+        uint256 id,
+        uint256 amount
+    ) public onlyRole(MINTER_ROLE) {
+        require(tokensTransferable, "NFT: Transfers by owner are disabled");
+        _safeTransferFrom(from, to, id, amount, "");
```
Comment (kristjanpeterson1) [inline contracts/ERC1155NftCustom.sol:80]: _safeTransferFrom works properly if msg.sender is not the same address that from yes?

**Review by kristjanpeterson1 — APPROVED** [review]

File: scripts/deploy-script.js [hunk 2]
```diff
@@ -3,9 +3,18 @@ const hre = require("hardhat");
...
+
+  // const ERC1155Tokens = await hre.ethers.getContractFactory("ERC1155NFT");
+  // const ERC1155tokens = await ERC1155Tokens.deploy("","Yo Block","YB");
+  // await ERC1155tokens.deployed();
+  // console.log("ERC1155NFT deployed to:", ERC1155tokens.address);
+
+  // const ERC1155TokensRoles = await hre.ethers.getContractFactory("ERC1155NFTRoles");
+  // const ERC1155tokensroles = await ERC1155TokensRoles.deploy("","Yo Block","YB","0xe30624B43A4C53a014b60db658928563a22A650A");
+  // await ERC1155tokensroles.deployed();
+  // console.log("ERC1155TokensRoles deployed to:", ERC1155tokensroles.address);
```
Comment (KasparPeterson) [inline scripts/deploy-script.js:17]: why are these commented out?

File: contracts/ERC1155NftCustom.sol [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+        uint256[] memory values
+    ) public onlyRole(MINTER_ROLE) {
+        require(tokensBurnable, "NFT: tokens burning is disabled");
+        _burnBatch(_owner, ids, values);
+        for (uint256 i = 0; i < ids.length; i++) {
+            tokenSupply[ids[i]] -= values[i];
+        }
+    }
+
+    function transferByOwner(
```
Comment (hyperspacebunny) [inline contracts/ERC1155NftCustom.sol:94]: Is this meant to transfer tokens only from the contract owner to another address or between any two addresses?

File: contracts/ERC1155NftCustom.sol [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+        }
+    }
+
+    function mintByOwner( address account, uint256 id, uint256 amount, string memory uri) public onlyRole(MINTER_ROLE) {
+        require(!_exists(id), "NFT: token already minted");
+        if (bytes(uri).length > 0) {
+            _tokenURIs[id] = uri;
+            emit URI(uri, id);
+        }
+        _mint(account, id, amount, "");
```
Comment (hyperspacebunny) [inline contracts/ERC1155NftCustom.sol:152]: Unlikely to be a major issue in this case but external interactions should happen last.

File: contracts/ERC1155NftCustom.sol [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+
+    function mintByOwnerBatch(
+        address[] memory to,
+        uint256[] memory ids,
+        uint256[] memory amounts,
+        string[] memory uris
+    ) public onlyRole(MINTER_ROLE) {
+        for (uint256 i = 0; i < ids.length; i++) {
+            require(!_exists(ids[i]), "NFT: one of tokens are already minted");
+            require(to[i] == address(to[i]), "NFT: one of addresses is invalid");
```
Comment (hyperspacebunny) [inline contracts/ERC1155NftCustom.sol:164]: Is this necessary? Wouldn't the txn revert anyway if the address isn't valid?

File: contracts/ERC1155NftCustom.sol (hunk 3 above)
Comment (kristjanpeterson1) [reply contracts/ERC1155NftCustom.sol:94]: only the _owner

**Pull request merged**